interceptors/unary_interceptors.go

```diff
@@ -43,7 +44,45 @@ func ResponseTimeLoggingInterceptor() grpc.UnaryServerInterceptor {
 	}
 }
 
-//NewRelicInterceptor intercepts all server actions and reports them to newrelic
+func GRPCLoggingInterceptor(enableRequestLog, enableResponseLog bool) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		// dont log for HTTP request, let HTTP Handler manage it
+		if !modifiers.IsHTTPRequest(ctx) {
+			defer func(begin time.Time) {
+				args := []interface{}{
+					"took",
+					time.Since(begin),
+				}
+				hasErr := err != nil
+				enableRequestLog = enableRequestLog || hasErr
+				enableResponseLog = enableResponseLog || hasErr
```

Comment:
this is a good idea - but if we design like this, probably we have to use `string` type instead of a `enum`

1. actual flag user is in `interceptors` pkg
2. the flag will be passed from `handlers` pkg
3. we cannot import `orion.Config` to `handlers` (otherwise cyclic import and required more refactor)

that means each package we need to define the same enum, which I think is too overkill
I suggest we just keep this for now (print resp/req if err), how do you think?

Reply:
I'm not in favor of changing the default behavior to log the request/response when there's an error.

1. Orion is open sourced, not an internal framework. We need to be mindful about not changing default behavior.
2. Logging request/response on errors is too opinionated. It's not something every service will want. The obvious example is PII data; unilaterally logging the request will risk accidentally logging PII data.

For opinionated changes like this, how about adding a new server interceptor to our `logging` package in go-utils? Your interceptor can use `AddToLogContext` to add the request/response to the log. The package has some other interceptors using this approach, which you can reference. Your services can then explicitly include the new interceptor if they want this behavior.